subworkflows/local/binning_refinement.nf

```diff
@@ -0,0 +1,116 @@
+
+workflow BINNING_REFINEMENT {
+    take:
+    contigs        //
+    bins           // channel: [ val(meta), path(bins) ]
+    depths
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Drop unnecessary files
+    ch_contigs_for_dastool = contigs
+                                .map {
+                                    meta, assembly, bams, bais ->
+                                        def meta_new = meta.clone()
+                                        [ meta_new, assembly ]
+                                }
+
+    // Rename for consistency for downstream steps (BUSCO etc.)
+    ch_bins_for_fastatocontig2bin = RENAME_PREDASTOOL(bins).renamed_bins
+                                        .map {
+                                            meta, bins ->
+                                                def meta_new = meta.clone()
+                                                meta_new.binner = meta.binner + 'Refined'
+
+                                                [ meta_new, bins ]
+                                        }
+                                        .branch {
+                                            metabat2: it[0]['binner'] == 'MetaBAT2Refined'
+                                            maxbin2:  it[0]['binner'] == 'MaxBin2Refined'
+                                        }
+
+    // Generate DASTool auxilary files
+    DASTOOL_FASTATOCONTIG2BIN_METABAT2 ( ch_bins_for_fastatocontig2bin.metabat2, "fa")
+    DASTOOL_FASTATOCONTIG2BIN_MAXBIN2 ( ch_bins_for_fastatocontig2bin.maxbin2, "fasta")
+
+    // Ru DAST
+    ch_fastatocontig2bin_for_dastool = Channel.empty()
+    ch_fastatocontig2bin_for_dastool = ch_fastatocontig2bin_for_dastool
+                                    .mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.fastatocontig2bin)
+                                    .mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.fastatocontig2bin)
+                                    .map {
+                                        meta, fastatocontig2bin ->
+                                            def meta_new = meta.clone()
+                                            meta_new.remove('binner')
+                                            [ meta_new, fastatocontig2bin ]
+                                    }
+                                    .groupTuple(by: 0)
+
+    ch_input_for_dastool = ch_contigs_for_dastool.join(ch_fastatocontig2bin_for_dastool, by: 0)
+
+    ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.versions.first())
+    ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.versions.first())
+
+    // Run DAStool
+    DASTOOL_DASTOOL(ch_input_for_dastool, [], [])
+
+    ch_dastool_bins_newmeta = DASTOOL_DASTOOL.out.bins.transpose()
+        .map {
+            meta, bins ->
+                def meta_new = meta.clone()
+
+                meta_new['binner'] = bins.name.split("-")[1]
+                [ meta_new, bins ]
```

Comment:
just trying to understand the code, this is actually only one bin here, right?
```suggestion
            meta, bin ->
                def meta_new = meta.clone()

                meta_new['binner'] = bin.name.split("-")[1]
                [ meta_new, bin ]
```

Reply:
Yes, after the transpose!